actors/builtin/miner/miner_actor.go

```diff
@@ -1751,7 +1757,13 @@ func handleProvingDeadline(rt Runtime) {
 			// Pledge for the sectors expiring early is retained to support the termination fee that will be assessed
 			// when the early termination is processed.
 			pledgeDelta = big.Sub(pledgeDelta, expired.OnTimePledge)
-			st.AddInitialPledgeRequirement(expired.OnTimePledge.Neg())
+			unlockedBalance = big.Add(unlockedBalance, expired.OnTimePledge)
+			st.AddInitialPledge(expired.OnTimePledge.Neg())
+
+			// Use unlocked pledge to pay down outstanding fee debt
+			feeToBurn := st.RepayPartialDebt(unlockedBalance)
+			unlockedBalance = big.Sub(unlockedBalance, feeToBurn) //nolint:ineffassign
+			penaltyTotal = big.Add(penaltyTotal, feeToBurn)
```

Comment:
Note there's a collision here with #1009, and I think that should probably land first.